src/utils/kubeclient.ts

```diff
@@ -53,3 +55,99 @@ export function createKubeClient(config: string | AppConfig, context?: string) {
 
   return kc;
 }
+
+function parseCanI(stdout: string, namespace: string): ClusterAccess {
+  const lines = stdout.split('\n');
+
+  const permissions: KubePermissions[] = [];
+  let hasFullAccess = false;
+
+  lines.forEach((line, index) => {
+    if (!index) {
+      return;
+    }
+    const columns = line.split(/\s{2,100}/);
+
+    /**
+     * an output line looks like this "selfsubjectrulesreviews.authorization.k8s.io [] [] [create]"
+     * and we need only the first and last items(resource name and verbs allowed)
+     */
+    const [resourceName, , , rawVerbs] = columns;
+
+    if (!resourceName) {
+      return;
+    }
+
+    const cleanVerbs = (rawVerbs as string).replace('[', '').replace(']', '');
+
+    if (resourceName === '*.*' && cleanVerbs === '*') {
+      hasFullAccess = true;
+    }
+
+    const verbs = cleanVerbs ? cleanVerbs.split(' ') : [];
+
+    permissions.push({
+      resourceName,
+      verbs,
+    });
+  });
+
+  return {
+    permissions,
+    hasFullAccess,
+    namespace,
+  };
+}
+
+export function getKubeAccess(namespace: string): ClusterAccess {
+  const command = `kubectl auth can-i --list --namespace=${namespace}`;
```

Comment:
this has to be run in the main thread using the runCommandInMainThread method to ensure the correct env